66_HLSLBxDFTests/app_resources/tests.hlsl

```diff
@@ -39,21 +74,23 @@ struct SBxDFTestResources
         SBxDFTestResources retval;
 
         retval.rng = nbl::hlsl::Xoroshiro64Star::construct(seed);
-        retval.u = float32_t3(rngFloat01(retval.rng), rngFloat01(retval.rng), 0.0);
-
-        retval.V.direction = nbl::hlsl::normalize<float32_t3>(rngFloat301(retval.rng));
-        retval.N = nbl::hlsl::normalize<float32_t3>(rngFloat301(retval.rng));
-        retval.T = nbl::hlsl::normalize<float32_t3>(rngFloat301(retval.rng));
+        retval.u = float32_t3(rngUniformDist<float32_t2>(retval.rng), 0.0);
 
-        retval.T = nbl::hlsl::normalize<float32_t3>(retval.T - nbl::hlsl::dot<float32_t3>(retval.T, retval.N) * retval.N); // gram schmidt
-        retval.B = nbl::hlsl::cross<float32_t3>(retval.N, retval.T);
+        retval.V.direction = nbl::hlsl::normalize<float32_t3>(projected_hemisphere_generate<float>(rngUniformDist<float32_t2>(retval.rng)));
+        retval.N = nbl::hlsl::normalize<float32_t3>(projected_hemisphere_generate<float>(rngUniformDist<float32_t2>(retval.rng)));
```

Comment:
not projected hemisphere, just hemisphere.

Also not hemisphere but full sphere, because its good to test that BRDFs handle invalid (below hemisphere) correctly